File: Proyectos/LeerImagen/002-escribirpixel.py
from PIL import Image
import time

# Start timer
start = time.time()

# Load the image
image = Image.open("Proyectos/LeerImagen/image.jpg").convert("RGB")
width, height = image.size
pixels = list(image.getdata())

# Convert to grayscale using median
def median_gray(rgb):
    return sorted(rgb)[1]  # median of R, G, B

gray_pixels = [median_gray(pixel) for pixel in pixels]

# Create a new grayscale image
gray_image = Image.new("L", (width, height))
gray_image.putdata(gray_pixels)
gray_image.save("imagenfinal.jpg", "JPEG", quality=100)

# End timer
elapsed = time.time() - start
print(f"Grayscale image saved as 'imagenfinal.jpg'")
print(f"Elapsed time: {elapsed:.4f} seconds")

File: Proyectos/LeerImagen/002-escribirpixel.cpp
#define STB_IMAGE_IMPLEMENTATION
#define STB_IMAGE_WRITE_IMPLEMENTATION

#include "stb_image.h"
#include "stb_image_write.h"

#include <iostream>
#include <vector>
#include <chrono>
#include <algorithm>

int main() {
    using clock = std::chrono::high_resolution_clock;
    auto start = clock::now();

    int width, height, channels;
    unsigned char* input_data = stbi_load("image.jpg", &width, &height, &channels, 0);
    if (!input_data) {
        std::cerr << "Failed to load image.\n";
        return 1;
    }

    std::cout << "Loaded image: " << width << "x" << height << ", channels: " << channels << "\n";

    std::vector<unsigned char> grayscale(width * height);

    for (int i = 0; i < width * height; ++i) {
        unsigned char r = input_data[i * channels + 0];
        unsigned char g = input_data[i * channels + 1];
        unsigned char b = input_data[i * channels + 2];

        // Median grayscale
        unsigned char rgb[] = { r, g, b };
        std::sort(rgb, rgb + 3);
        grayscale[i] = rgb[1];  // median
    }

    // Write grayscale image (1 channel)
    if (!stbi_write_jpg("imagenfinal.jpg", width, height, 1, grayscale.data(), 100)) {
        std::cerr << "Failed to write grayscale image.\n";
        stbi_image_free(input_data);
        return 1;
    }

    stbi_image_free(input_data);

    auto end = clock::now();
    std::chrono::duration<double> elapsed = end - start;

    std::cout << "Grayscale image written to 'imagenfinal.jpg'.\n";
    std::cout << "Elapsed time: " << elapsed.count() << " seconds.\n";

    return 0;
}
